stdlib/src/collections/vector.mojo

```diff
@@ -125,37 +125,47 @@ struct InlinedFixedVector[
         self.current_size = 0
         self.capacity = capacity
 
-    # TODO: Probably don't want this to be implicitly no-op copyable when we
-    # have ownership.
     @always_inline
-    fn __copyinit__(inout self, existing: Self):
-        """Creates a shallow copy (doesn't copy the underlying elements).
+    fn __init__(inout self, existing: Self):
+        """
+        Copy constructor.
 
         Args:
             existing: The `InlinedFixedVector` to copy.
         """
         self.static_data = existing.static_data
-        self.dynamic_data = existing.dynamic_data
+        self.dynamic_data = UnsafePointer[type]()
+        if existing.dynamic_data:
+            var ext_len = existing.capacity - size
+            self.dynamic_data = UnsafePointer[type].alloc(ext_len)
+            memcpy(self.dynamic_data, existing.dynamic_data, ext_len)
+
         self.current_size = existing.current_size
         self.capacity = existing.capacity
 
     @always_inline
-    fn _del_old(self):
-        """Destroys the object."""
-        if self.capacity > Self.static_size:
-            self.dynamic_data.free()
+    fn __moveinit__(inout self, owned existing: Self):
+        """
+        Move constructor.
 
-    @always_inline
-    fn deepcopy(self) -> Self:
-        """Creates a deep copy of this vector.
+        Args:
+            existing: The `InlinedFixedVector` to consume.
+        """
+        self.static_data = existing.static_data
+        self.dynamic_data = existing.dynamic_data
+        self.current_size = existing.current_size
+        self.capacity = existing.capacity
 
-        Returns:
-            The created copy of this vector.
+        existing.dynamic_data = UnsafePointer[type]()
+
+    @always_inline
+    fn __del__(owned self):
         """
-        var res = Self(self.capacity)
-        for i in range(len(self)):
-            res.append(self[i])
-        return res
+        Destructor.
+        """
+        if self.dynamic_data:
+            self.dynamic_data.free()
+            self.dynamic_data = UnsafePointer[type]()
```

Comment:
**Question** Can you help me understand the motivation of nulling out this field?  For comparison, we don't do this in `List.__del__` at https://github.com/modularml/mojo/blob/80a9de7261fbe850fa5ceb37c1489052f18364ac/stdlib/src/collections/list.mojo#L211

Reply:
This isn't necessary in "correct" code, but it _does_ prevent even incorrect usage of things like MaybeUninit double destruction from causing issues since it turns a double free into a noop. Just mitigates damage in those scenarios with ~no cost

Reply:
(I'm more than happy to cut that line since that's probably an unnecessary amount of paranoia)